### PR TITLE
Add ngrx

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/app/user/user.module.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    IconsModule,
+    InputsModule,
+    InputUtilitiesModule,
+    WavesModule,
+    ReactiveFormsModule,
+    RouterModule,
+    MatFormFieldModule,
+    MatDatepickerModule,
+    MatNativeDateModule,
+    FormsModule
```
Comment (Ozzorany) [inline src/app/user/user.module.ts]: add MatInputModule

File: src/app/components/navbar/navbar.component.ts [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
-      this.signInUser.firstName.concat(' ', this.signInUser.lastName) : '';
+  ngOnInit() {
+    this.signInUser$ = this.store.pipe(select(getUser));
+    this.userFullName$ = this.signInUser$.pipe(map(user => user ? `${user.firstName} ${user.lastName}` : 'no name'));
   }
 
-  logout(): void {
-    this.authenticateService.logout();
-    this.router.navigateByUrl('/');
+  logout() {
```
Comment (Ozzorany) [inline src/app/components/navbar/navbar.component.ts]: why removed return type?

File: src/app/components/navbar/navbar.component.ts [hunk 3]
```diff
@@ -1,29 +1,30 @@
...
-  constructor(private router: Router,
-              private authenticateService: AuthenticateService) {
+  constructor(private router: Router, private store: Store<UserState>) {
   }
 
-  ngOnInit(): void {
-    this.signInUser = this.authenticateService.getUser();
-    this.userFullName = !!this.signInUser ?
-      this.signInUser.firstName.concat(' ', this.signInUser.lastName) : '';
+  ngOnInit() {
```
Comment (Ozzorany) [inline src/app/components/navbar/navbar.component.ts]: why removed return type?

File: src/app/items/services/items.service.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
 import { Item } from 'src/entities/item.model';
-import { FirebaseService } from '../firebase/firebase.service';
-import { CollectionType } from '../firebase/models/collection-type.model';
-import { IFirebaseService } from '../firebase/models/firebase-service.interface';
+import { FirebaseService } from '../../services/firebase/firebase.service';
+import { CollectionType } from '../../services/firebase/models/collection-type.model';
+import { IFirebaseService } from '../../services/firebase/models/firebase-service.interface';
 
 @Injectable({
   providedIn: 'root'
```
Comment (Ozzorany) [inline src/app/items/services/items.service.ts]: remove this providedIn root , you set this provider already in items.module

**Review by Ozzorany — CHANGES_REQUESTED** [review]

File: src/app/user/actions/user.actions.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+export const login = createAction(userActionNames.LOGIN, props<{ email: string; password: string; }>());
+export const loginSuccess = createAction(userActionNames.LOGIN_SUCCESS, props<{ user: User }>());
+export const loginFail = createAction(userActionNames.LOGIN_FAIL, props<{ message: string; }>());
+
+export const logout = createAction(userActionNames.LOGOUT);
+export const logoutSuccess = createAction(userActionNames.LOGOUT_SUCCESS);
+export const logoutFail = createAction(userActionNames.LOGOUT_FAIL, props<{ message: string; }>());
+
+export const register = createAction(userActionNames.REGISTER,
+  props<{ user: Omit<User, 'id'> & { password: string; } }>()
```
Comment (yarivg) [inline src/app/user/actions/user.actions.ts:25]: what does it do? - & { password: string; } 
I got the part that omits the id key from the User.

File: src/app/user/components/sign-in/sign-in.component.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  ngOnInit() {
+    this.loginForm = new FormGroup({
+      passwordForm: new FormControl(null, [Validators.required, Validators.minLength(8)]),
+      emailForm: new FormControl(null, [Validators.required, Validators.email]),
+    });
+  }
+
+  signIn() {
```
Comment (yarivg) [inline src/app/user/components/sign-in/sign-in.component.ts]: add return types

File: src/app/user/effects/user.effects.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+      )),
+    )
+  );
+
+  constructor(
+    private actions$: Actions,
+    private authService: AuthenticateService,
+    private userService: UsersService,
+    private router: Router,
+    public afAuth: AngularFireAuth
```
Comment (yarivg) [inline src/app/user/effects/user.effects.ts]: rename to angularFireAuth

File: src/app/user/effects/user.effects.ts [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  signUp$ = createEffect(() => this.actions$.pipe(
+    ofType(register),
+    switchMap(({user}) => this.authService.signUp(user.phoneNumber)
+      .pipe(
+        switchMap(([confirmationResult, code]) =>
+          this.authService.verify(confirmationResult, code, user.email, user.password)),
+        switchMap(() => this.authService.getFirebaseCurrentUser$().pipe(
+          switchMap(({uid}) => this.userService.upsert$({
+          id: uid,
+          ...omit(user, 'password')
```
Comment (yarivg) [inline src/app/user/effects/user.effects.ts:85]: Isn't the password store in firebase DB? why do you omit it

File: src/app/user/services/authentication.service.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+  modalRef: MDBModalRef;
+  private countryDialogCode = '+972';
+  private userKey = 'USER';
+
+
+  constructor(private windowService: WindowService, private modalService: MDBModalService, private afAuth: AngularFireAuth) {
+    this.windowRef = windowService.windowRef;
+  }
+
+  signUp(phoneNumber: string): Observable<[ConfirmationResult, string]> {
```
Comment (yarivg) [inline src/app/user/services/authentication.service.ts]: signUp$

Comment (yarivg) [reply src/app/user/services/authentication.service.ts]: and all the other functions too

**Review by yarivg — CHANGES_REQUESTED** [review]
Good work

File: src/app/user/actions/user.actions.ts (hunk 5 above)
Comment (ghost) [reply src/app/user/actions/user.actions.ts:25]: the User interface doesn't have the user password in it
while registering I want to get all the user details and the password.
This type requires the password as well, as part of the object

File: src/app/user/effects/user.effects.ts (hunk 8 above)
Comment (ghost) [reply src/app/user/effects/user.effects.ts:85]: The password is managed by the Authentication DB. We don't want to store it in the user itself.

**Review by Ozzorany — APPROVED** [review]

**Pull request merged**